x/escrow/keeper/keeper.go

```diff
@@ -61,25 +68,50 @@ func (k *keeper) AccountCreate(ctx sdk.Context, id types.AccountID, owner sdk.Ac
 		ID:          id,
 		Owner:       owner.String(),
 		State:       types.AccountOpen,
-		Balance:     deposit,
+		Balance:     sdk.NewCoin(deposit.Denom, sdk.ZeroInt()),
 		Transferred: sdk.NewCoin(deposit.Denom, sdk.ZeroInt()),
 		SettledAt:   ctx.BlockHeight(),
+		Depositor:   depositor.String(),
+		Funds:       sdk.NewCoin(deposit.Denom, sdk.ZeroInt()),
 	}
 
 	if err := obj.ValidateBasic(); err != nil {
 		return err
 	}
 
-	if err := k.bkeeper.SendCoinsFromAccountToModule(ctx, owner, types.ModuleName, sdk.NewCoins(deposit)); err != nil {
-		return err
+	if owner.Equals(depositor) {
+		// use owner's account to add balance to escrow account
+		if err := k.bkeeper.SendCoinsFromAccountToModule(ctx, owner, types.ModuleName, sdk.NewCoins(deposit)); err != nil {
+			return err
+		}
+		obj.Balance = deposit
+	} else {
+		if err := k.fetchFundsFromDepositor(ctx, owner, depositor, deposit); err != nil {
+			return err
+		}
+		obj.Funds = deposit
```

Comment:
I wonder if we're going to want to be able to deposit from both accounts, with different amounts for each.  We can add a flag for this in the future If we want to.